simperby/src/node/state_storage.rs

```diff
@@ -32,46 +122,90 @@ impl StateStorage {
     /// Assuming an empty storage, creates the genesis state.
     pub async fn genesis(
         mut storage: Box<dyn KVStorage>,
-        genesis_info: GenesisInfo,
+        _genesis_info: GenesisInfo,
     ) -> Result<Self, Error> {
-        unimplemented!()
+        insert_or_update_json_to_storage(
+            storage.as_mut(),
+            create_delegation_state_key(),
+            &DelegationState {
+                original_validator_set: Default::default(),
+            },
+        )
+        .await?;
+        storage
+            .insert_or_update(create_state_root(), Hash256::zero().as_ref())
+            .await?;
+        Ok(Self { storage })
     }
 
     /// Assuming an existing state, check the basic validity of the state.
     pub async fn new(storage: Box<dyn KVStorage>) -> Result<Self, Error> {
```

Comment:
Is it the comment and function name and implementation that you've intended?

Reply:
Yes. This is a c.f. to `genesis()`. I will elaborate the description soon. (I have next PR)